x-pack/plugins/search_playground/public/plugin.ts

```diff
@@ -39,17 +40,21 @@ export class SearchPlaygroundPlugin
     deps: AppPluginSetupDependencies
   ): SearchPlaygroundPluginSetup {
     if (!this.config.ui?.enabled) return {};
+    const featureFlags = {
+      searchPlaygroundEnabled: isSearchModeEnabled(core.uiSettings),
+    };
 
     core.application.register({
       id: PLUGIN_ID,
-      appRoute: '/app/search_playground',
+      appRoute: PLUGIN_PATH,
       title: PLUGIN_NAME,
       async mount({ element, history }: AppMountParameters) {
         const { renderApp } = await import('./application');
         const [coreStart, depsStart] = await core.getStartServices();
         const startDeps: AppPluginStartDependencies = {
           ...depsStart,
           history,
+          featureFlags,
```

Comment:
Putting this `featureFlags` object in the kibana services will only work in serverless. Since this mount is only used in ES3. In stack `enterprise_search` renders the page (so that we can have the side nav) and the `useKibana` hook will provide the services defined in `enterprise_search` application mount. Which won't have this object.

Did you test this for stack? if we don't get a runtime error then _maybe_ that ok and we could only test this in serverless?

Another way to solve this would be to introduce a hook just for the feature flag which does the `uiSettings` check in the hook since `uiSettings` will always be there for both stack and serverless.

Something like:

```typescript
// hooks/use_search_playground_feature_flag.ts
import { isSearchModeEnabled } from '../utils/feature_flags';
import { useKibana } from './use_kibana';

export const useSearchPlaygroundFeatureFlag = (): boolean => {
  const { uiSettings } = useKibana().services;
  
  return uiSettings ? isSearchModeEnabled(uiSettings) : false;
};
```

Reply:
>Putting this featureFlags object in the kibana services will only work in serverless. Since this mount is only used in ES3. In stack enterprise_search renders the page (so that we can have the side nav) and the useKibana hook will provide the services defined in enterprise_search application mount. Which won't have this object.

Yes. This is humble attempt to not create a giant PR when the whole feature is implemented, but rather smaller safer increments that doesn't create issues in `serverless` scope.

I tested it on stack and it doesn't work but also not failing at runtime. It was my intention, which came out after a few rounds of review, i didn't like the way it is very different in between both. I saw the `uiSettings` is available in `KibanaLogic` in stack, so what you suggested is where I wanted to go eventually but didn't came up with the solution yet. But hey you saved me from typing more haha :D 

TL;DR Intentionally not implemented in stack, was looking for solutions. Now I have it , I will apply. <3